README.md

```diff
@@ -1 +1,263 @@
+```
+
+Run multiple `package.json` `scripts` tasks.
+
+```sh
+$ builder concurrent foo-task bar-task baz-task
+```
+
+
+## Tasks
+
+The underyling concept here is that `builder` `script` commands simply _are_
+NPM-friendly `package.json` `script` commands. Pretty much anything that you
+can execute with `npm run FOO` can be executed with `builder run FOO`.
+
+Builder can run 1+ tasks based out of `package.json` `scripts`. For a basic
+scenario like:
+
+```js
+{
+  "scripts": {
+    "foo": "echo FOO",
+    "bar": "echo BAR"
+  }
+}
+```
+
+Builder can run these tasks individually:
+
+```sh
+$ builder run foo
+$ builder run bar
+```
+
+Sequentially via `||` or `&&` shell helpers:
+
+```sh
+$ builder run foo && builder run bar
+```
+
+Concurrently via the Builder built-in `concurrent` command:
+
+```sh
+$ builder concurrent foo bar
+```
+
+With `concurrent`, all tasks continue running until they all complete _or_
+any task exits with a non-zero exit code, in which case all still alive tasks
+are killed and the Builder process exits with the error code.
+
+## Archetypes
+
+Archetypes deal with common scenarios for your projects. Like:
+
+* [builder-react-component][]: A React component
+* A React application server
+* A Chai / jQuery / VanillaJS widget
+
+Archetypes typically provide:
+
+* A `package.json` with `builder`-friendly `script` tasks.
+* Dependencies and dev dependencies to build, test, etc.
+* Configuration files for all `script` tasks.
+
+In most cases, you won't need to override anything. But, if you do, pick the
+most granular `scripts` command in the archetype you need to override and
+define _just that_ in your project's `package.json` `script` section. Copy
+any configuration files that you need to tweak and re-define the command.
+
+### Task Resolution
+
+The easiest bet is to just have _one_ archetype per project. But, multiple are
+supported. In terms of `scripts` tasks, we end up with the following example:
+
+```
+ROOT/package.json
+ROOT/node_modules/ARCHETYPE_ONE/package.json
+ROOT/node_modules/ARCHETYPE_TWO/package.json
+```
+
+Say we have a `.builderrc` like:
+
+```yaml
+---
+archetypes:
+  - ARCHETYPE_ONE
+  - ARCHETYPE_TWO
+```
+
+The resolution order for a `script` task (say, `foo`) present in all three
+`package.json`'s would be the following:
+
+* Look through `ROOT/package.json` then the configured archetypes in _reverse_
+  order: `ARCHETYPE_TWO/package.json`, then `ARCHETYPE_ONE/package.json` for
+  a matching task `foo`
+* If found `foo`, check if it is a "passthrough" task, which means it delegates
+  to a later instance -- basically `"foo": "builder run foo"`. If so, then look
+  to next instance of task found in order above.
+
+### Special Archetype Tasks
+
+Archetypes use conventional `scripts` task names, except for the following
+special cases:
+
+* `"npm:postinstall"`
+* `"npm:preversion"`
+* `"npm:version"`
+* `"npm:test"`
+
+These tasks are specifically actionable during the `npm` lifecycle, and
```

Comment:
Not doing it this way in `bolt` caused me a lot of headache...this is a great convention.